### PR TITLE
Shelley consensus integration

### PR DESCRIPTION
**Pull request by nc6**
Review commit by commit

### CODE REVIEW

File: cardano-crypto-class/src/Cardano/Crypto/DSIGN/Ed448.hs [hunk 1]
```diff
@@ -8,20 +8,25 @@
 
 -- | Ed448 digital signatures.
 module Cardano.Crypto.DSIGN.Ed448
```
Comment (mrBliss) [inline cardano-crypto-class/src/Cardano/Crypto/DSIGN/Ed448.hs]: Hmm, it seems this file has been reformatted. Can you undo those changes?

File: cardano-crypto-class/src/Cardano/Crypto/Hash/Class.hs [hunk 2]
```diff
@@ -6,8 +6,7 @@
 module Cardano.Crypto.Hash.Class
   ( HashAlgorithm (..)
   , ByteString
-  , Hash
-  , getHash
+  , Hash(..)
```
Comment (mrBliss) [inline cardano-crypto-class/src/Cardano/Crypto/Hash/Class.hs]: For consistency with the rest of the export list: add a space between `Hash` and `(..)`.

File: slotting/src/Cardano/Slotting/SlotBounded.hs [hunk 3]
```diff
@@ -61,6 +66,23 @@ data SlotBounded (bounds :: Bounds) a
...
+      <*> fromCBOR
+
+instance (ToCBOR a, Typeable b) => ToCBOR (SlotBounded b a) where
+  toCBOR SlotBounded {sbLower, sbUpper, sbContent} =
+    mconcat
+      [ CBOR.encodeListLen 3,
+        toCBOR sbLower,
+        toCBOR sbUpper,
+        toCBOR sbContent
+      ]
```
Comment (mrBliss) [inline slotting/src/Cardano/Slotting/SlotBounded.hs]: I don't know what the convention is in this repo, but in ouroboros-consensus, we start lines with commas.

Comment (dnadales) [reply slotting/src/Cardano/Slotting/SlotBounded.hs]: :heavy_plus_sign:

File: slotting/src/Cardano/Slotting/SlotBounded.hs [hunk 4]
```diff
@@ -61,6 +66,23 @@ data SlotBounded (bounds :: Bounds) a
       }
   deriving (Eq, Functor, Show, Generic, Serialise, NoUnexpectedThunks)
 
+instance (FromCBOR a, Typeable b) => FromCBOR (SlotBounded b a) where
```
Comment (dnadales) [inline slotting/src/Cardano/Slotting/SlotBounded.hs]: Don't we need roundtrip tests for these?

**Pull request merged**